cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/TimePartitionedStreamFileWriter.java

```diff
@@ -162,7 +167,25 @@ public int hashCode() {
       }
 
       LOG.debug("New stream file created at {}", eventFile);
-      return new StreamDataFileWriter(createOutputSupplier(eventFile), createOutputSupplier(indexFile), indexInterval);
+
+      final StreamDataFileWriter streamDataFileWriter =
+        new StreamDataFileWriter(createOutputSupplier(eventFile), createOutputSupplier(indexFile), indexInterval);
+
+      timer.schedule(new TimerTask() {
+        @Override
+        public void run() {
+          try {
+            // NOTE: this doesn't remove it from PartitionedFileWriter map of writers
+            // it will get removed from there when the next event comes
+            streamDataFileWriter.close();
+          } catch (IOException e) {
+            Throwables.propagate(e);
```

Comment:
Should we propagate this exception if we are not able to close the stream files?